src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    finalizeClusterFor(pileup, Seq(), pileupToNeighbors)
+  }
+
+  /**
+   * Similar to `extractClusterFor()`, but picking up from a partially complete recursive call stack.
+   * Implementation note: the implementation is not tail-recursive, but in practice the recursion depth is small.
+   * @param pileup The pileup representative of the cluster that will be returned
+   * @param existingCluster Partially complete cluster that the given pileup belongs to
+   * @param remainingNeighborGroups Additional pileup-to-neighbor-set relationships to examine and possibly add to the
+   *                                growing cluster
+   * @return (1) The complete cluster for the given pileup; (2) Remaining map of pileup to neighbor set
+   */
+  private def finalizeClusterFor(
+                          pileup: BreakpointPileup,
+                          existingCluster: PileupGroup,
+                          remainingNeighborGroups: Map[BreakpointPileup, PileupGroup]):
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    if (existingCluster.contains(pileup) && !remainingNeighborGroups.contains(pileup)) {
+      (existingCluster, remainingNeighborGroups)
+    } else {
+      val neighbors: PileupGroup = remainingNeighborGroups(pileup)
+      val reducedRemainingGroups: Map[BreakpointPileup, PileupGroup] = remainingNeighborGroups.removed(pileup)
+      if (reducedRemainingGroups.isEmpty) (neighbors :+ pileup, Map())
+      else neighbors.foldLeft[(PileupGroup, Map[BreakpointPileup, PileupGroup])](
+        (neighbors :+ pileup, reducedRemainingGroups))(
+        (currClusterAndRemaining: (PileupGroup, Map[BreakpointPileup, PileupGroup]), nextNeighbor: BreakpointPileup) => {
+          currClusterAndRemaining match {
+            case (currCluster, remaining) =>
+              finalizeClusterFor(nextNeighbor, currCluster, remaining) match {
+                case (expandedCluster, reducedRemaining) => ((currCluster ++ expandedCluster).distinct, reducedRemaining)
+              }
+          }
+        }
+      )
+    }
+  }
+
+}
+
+
+/** Contig and strand for the left and right sides of a breakpoint pileup */
+case class BreakpointContigOrientation(left_contig: String, left_strand: Char, right_contig: String, right_strand: Char)
+
+
+object BreakpointContigOrientation {
+
+  def apply(pileup: BreakpointPileup): BreakpointContigOrientation = {
+    BreakpointContigOrientation(
+      left_contig  = pileup.left_contig,
+      left_strand  = pileup.left_strand,
+      right_contig = pileup.right_contig,
+      right_strand = pileup.right_strand,
+    )
+  }
+
+}
+
+
+/** Type of structural variant supported by a pileup */
+object BreakpointCategory extends Enumeration {
+  type BreakpointCategory = String
+  val POSSIBLE_DELETION = "Possible deletion"
+  val POSSIBLE_INTRA_CONTIG = "Possible intra-contig rearrangement"
+  val POSSIBLE_INTER_CONTIG = "Possible inter-contig rearrangement"
```

Comment:
No shouting case please!  In scala we use `TitleCase` for constants.

Reply:
I think we can drop the "possible" from the latter two.  I thought we wanted to call the first one possible/likely etc. as we can't be sure it's a deletion and not something else going on, but the latter two are broad enough that they are effectively always correct.